operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -516,6 +521,33 @@ type ScannerComponentSpec struct {
 	Monitoring *Monitoring `json:"monitoring,omitempty"`
 }
 
+// ScannerV4ComponentSpec defines settings for the central "scannerV4" component.
+type ScannerV4ComponentSpec struct {
+	// If you want to deploy the Red Hat Advanced Cluster Security ScannerV4, you can enable it here
+	// By default, the ScannerV4 is disabled.
+	//+kubebuilder:default=Disabled
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1,displayName="Scanner V4 Component"
+	ScannerComponent *ScannerComponentPolicy `json:"scannerComponent,omitempty"`
+
+	// Settings pertaining to the indexer deployment.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:fieldDependency:.scannerComponent:Enabled"}
+	Indexer *ScannerV4Component `json:"indexer,omitempty"`
```

Comment:
I am wondering if we should use different types for `Indexer` for `Matcher`. I could well imagine that there will be component-specific configuration options soon?

Reply:
I had it specific at first, but noticed that I had to duplicate lots of code for that. Since the config shapes are both exactly the same and the operator exposes even less configuration options, I thought it's unlike that we'll expose different configuration anytime soon.

Reply:
Since it can always be changed when need arises I am fine with this.
FYI: @RTann, @dcaravel.